_plugins/show.rb

```diff
@@ -31,60 +52,106 @@ def get_playwright_formatted_long(show)
       return ret
     end
 
-    # Main generation method
-    def generate(site)
-      puts "Processing shows..."
-      all_shows = site.collections["shows"].docs
-      @years = site.collections["years"].docs
-      @people = site.collections["people"].docs
+    def get_show_override_assets(show)
+      show.data["assets"].select { |i| i["display_image"] == true }
+    end
 
-      # This is here as it's needed here, I expect year.rb to be run after.
+    def get_show_asset_type(show, key)
+      show.data["assets"].select { |i|
+        if i.has_key?("page")
+          i["type"] == key and not i["page"] > 1
+        else
+          i["type"] == key
+        end
+      }
+    end
 
-      years_by_slug = generate_years_by_slug(@years)
-      people_by_filename = generate_people_by_filename(@people)
+    def get_show_display_image(show)
+      # Assets required
+      return nil if not show.data.has_key?("assets")
```

Comment:
`Hash#has_key?` is deprecated in favor of `Hash#key?`.<br>Favor `unless` over `if` for negative conditions.<br>Use `!` instead of `not`.